### ISSUE TITLE
Support for non-saturating mode for fp8

### ISSUE THREAD

**Issue by wonjeon** [jax-ml/ml_dtypes#147]
OFP8 (https://www.opencompute.org/documents/ocp-8-bit-floating-point-specification-ofp8-revision-1-0-2023-12-01-pdf-1) defines two saturation modes - saturating and non-saturating mode.
It looks like the current code runs in saturating mode by default.
```
>>> math.isnan(np.float32(449).astype(float8_e4m3fn))
False
```
Wondering if there's any plan to support both.
Thank you.

**Comment by awstobias** [comment]
No, the default is actually non-saturating. What you are seeing are effects of rounding. Try a larger float32 and you'll see it produces NaN.
